app/models/goal.py

```diff
@@ -1,5 +1,25 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from typing import Optional
 from ..db import db
 
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    tasks: Mapped[list["Task"]] = relationship(back_populates="goal")
+
+    def to_dict(self):
+        goal_as_dict = {
+            "id": self.id,
+            "title": self.title
+        }
+
+        return goal_as_dict
```

Comment:
Could also directly return a dictionary literal since `goal_as_dict` gets created and then immediately returned.

```suggestion
        return {
            "id": self.id,
            "title": self.title
        }
```